lib/index.js

```diff
@@ -488,11 +482,29 @@ MapboxGeocoder.prototype = {
           this.eventManager.start(this);
           this.fresh = false;
         }
-        // supplement Mapbox Geocoding API results with locally populated results
-        res.features = res.features
-          ? localGeocoderRes.concat(res.features)
-          : localGeocoderRes;
 
+        if (this.options.localGeocoder) {
+
+          localGeocoderRes = this.options.localGeocoder(searchInput, res.features) || [];
+
+          // supplement Mapbox Geocoding API results with features returned by a promise
+          if(typeof localGeocoderRes.then === 'function') {
+            return localGeocoderRes.then(function(features) {
+              res.features = res.features ? features.concat(res.features) : features;
+              return res;
+            }, function(){
+              // on error, display the original result
+              return res;
+            });
+          } else {
+            // supplement Mapbox Geocoding API results with locally populated results
+            res.features = res.features ? localGeocoderRes.concat(res.features) : localGeocoderRes;
```

Comment:
I think where the localGeocoder returns a result faster than the Geocoding API (which it should if it's populated from a local source) then it should display the results immediately (like it currently should) instead of waiting for the Geocoding API to return.